core/indexer/state.go

```diff
@@ -2,28 +2,42 @@ package indexer
 
 import (
 	"context"
+	"encoding/binary"
 	"errors"
 
 	"github.com/Layr-Labs/eigenda/core"
 	"github.com/Layr-Labs/eigenda/indexer"
+	lru "github.com/hashicorp/golang-lru/v2"
 )
 
 type IndexedChainState struct {
 	core.ChainState
 
 	Indexer indexer.Indexer
+
+	operatorStateCache *lru.Cache[string, *core.IndexedOperatorState]
 }
 
 var _ core.IndexedChainState = (*IndexedChainState)(nil)
 
 func NewIndexedChainState(
 	chainState core.ChainState,
 	indexer indexer.Indexer,
+	cacheSize int,
 ) (*IndexedChainState, error) {
-
+	operatorStateCache := (*lru.Cache[string, *core.IndexedOperatorState])(nil)
+	var err error
+	if cacheSize > 0 {
+		operatorStateCache, err = lru.New[string, *core.IndexedOperatorState](cacheSize)
```

Comment:
2 questions:
1. Should we set a max cache size based on available memory etc?
2. Is there a default expiry of the cache ?